presto-hive/src/main/java/com/facebook/presto/hive/ColumnarBinaryHiveRecordCursor.java

```diff
@@ -636,6 +641,9 @@ private void parseColumn(int column)
         else if (BIGINT.equals(type)) {
             parseLongColumn(column);
         }
+        else if (INTEGER.equals(type)) {
+            parseLongColumn(column);
```

Comment:
That ends up calling into `readVInt` which already has a loop and a branch, so I would guess it's not a problem. I'd ask @dain's opinion though, since he has more experience with the readers